packages/upset/src/components/Columns/BookmarkStar.tsx

```diff
@@ -1,28 +1,134 @@
 import { Row } from '@visdesignlab/upset2-core';
-import { FC } from 'react';
-import { useRecoilValue } from 'recoil';
 import StarIcon from '@mui/icons-material/Star';
+import {
+  FC,
+  MouseEvent,
+  useContext,
+  useMemo,
+  useState,
+} from 'react';
+import { useRecoilValue } from 'recoil';
+import { ProvenanceContext } from '../Root';
 import { dimensionsSelector } from '../../atoms/dimensionsAtom';
 import translate from '../../utils/transform';
-import { bookmarkedColorPalette } from '../../atoms/config/currentIntersectionAtom';
+import {
+  bookmarkedColorPalette,
+  bookmarkSelector,
+  nextColorSelector,
+} from '../../atoms/config/currentIntersectionAtom';
 
 type Props = {
-    row: Row;
-}
+  row: Row;
+};
 
+const BASE_OPACITY = 0.0;
+const HOVERED_OPACITY = 0.5;
+const BOOKMARKED_OPACITY = 1.0;
+
+/**
+ * BookmarkStar component renders a star icon that can be bookmarked.
+ * It changes its color and opacity based on the bookmark and hover states.
+ *
+ * @component
+ * @param {Props} props - The properties object.
+ * @param {Object} props.row - The row data for the current item.
+ * @returns {JSX.Element} The rendered BookmarkStar component.
+ *
+ * @example
+ * <BookmarkStar row={row} />
+ */
 export const BookmarkStar: FC<Props> = ({ row }) => {
   const dimensions = useRecoilValue(dimensionsSelector);
   const colorPallete = useRecoilValue(bookmarkedColorPalette);
+  const nextColor = useRecoilValue(nextColorSelector);
+  const bookmarks = useRecoilValue(bookmarkSelector);
+  const { actions } = useContext(ProvenanceContext);
+
+  const [hovered, setHovered] = useState(false);
+  const bookmarked = useMemo(() => bookmarks.find((b) => b.id === row.id), [
```

Comment:
I think you should change this to a Recoil selector for performance; amongst other improvements, that means this component no longer needs to subscribe to `bookmarks` and re-render every time it changes.

Reply:
Resolved in https://github.com/visdesignlab/upset2/pull/416/commits/d550fa27b74b06e59de873b0a59879abd4cfc9f7

Agree that this is probably a negligible performance impact, as the selectors still are reactive to the `bookmarks` selector, but better to centralize if reused.